engine/src/flutter/lib/ui/painting/fragment_program.cc

```diff
@@ -58,8 +59,15 @@ std::string FragmentProgram::initFromAsset(const std::string& asset_name) {
            std::string("' does not contain any shader data.");
   }
 
-  impeller::RuntimeStageBackend backend =
-      ui_dart_state->GetRuntimeStageBackend();
+  auto impeller_context =
+      ui_dart_state->GetIOManager()->GetImpellerContext().get();
```

Comment:
This is failing an `fml::WeakPtr` dereference assertion because the `IOManager` can not be safely used on the UI thread.

Reply:
ahh ok. I think we can use a different future on ui_dart_state